### ISSUE TITLE
UICollectionViewFlowLayoutBreakForInvalidSizes with ButtonBarView

### ISSUE THREAD

**Issue by miranda-elliott** [xmartlabs/XLPagerTabStrip#545]
I'm getting this error:
```
The behavior of the UICollectionViewFlowLayout is not defined because:
the item height must be less than the height of the UICollectionView minus the section insets top and bottom values, minus the content insets top and bottom values.
The relevant UICollectionViewFlowLayout instance is <UICollectionViewFlowLayout: 0x7f976b54f870>, and it is attached to <XLPagerTabStrip.ButtonBarView: 0x7f976c10b000; baseClass = UICollectionView; frame = (-19.5 0; 414 64); clipsToBounds = YES; autoresize = RM+BM; gestureRecognizers = <NSArray: 0x600000446840>; layer = <CALayer: 0x600000230280>; contentOffset: {0, -24}; contentSize: {414, 64}; adjustedContentInset: {24, 0, 0, 0}> collection view layout: <UICollectionViewFlowLayout: 0x7f976b54f870>.
Make a symbolic breakpoint at UICollectionViewFlowLayoutBreakForInvalidSizes to catch this in the debugger.
```

When I make a symbolic breakpoint, it fails on line 81 of ButtonBarView.swift in the `move(fromIndex: Int, toIndex: Int, progressPercentage: CGFloat, pagerScroll: PagerScroll)` method where `let fromFrame = layoutAttributesForItem(at: IndexPath(item: fromIndex, section: 0))!.frame`. 

Here are my layout constraints:
Height of 64. Top, leading, and trailing to the superview. Bottom to the container view.
<img width="547" alt="Interface Builder Screenshot" src="https://user-images.githubusercontent.com/8364014/36645250-cd92ff68-1a33-11e8-8476-90e89ef501fe.png">

What am I doing wrong? I can't figure out what is causing this error and whether it's my fault or a problem with ButtonBarView. Thank you!

**Comment by tooolbox** [comment]
I am having this same problem.  It seems like it's been going on for a few months with no resolution, based on my searches through the Issues.

@santiagofm Any thoughts?

**Comment by tooolbox** [comment]
Whoops.

Turns out I just read the readme incorrectly, adding a **UIView** to Interface Builder instead of a **UICollectionView**.  I had set the class to `ButtonBarView` but I guess a difference in how these two items are set up in Interface Builder was causing my issue.  Reading through [the referenced community blog post](https://medium.com/michaeladeyeri/how-to-implement-android-like-tab-layouts-in-ios-using-swift-3-578516c3aa9) was what pointed this out to me.

Maybe @miranda-elliott you're having the same problem, maybe not.

**Comment by miranda-elliott** [comment]
Unfortunately I am already using a UICollectionView - also followed the steps in the Medium post you linked @tooolbox.

The weird part is that the button bar looks strange on initial load:
![img_0980](https://user-images.githubusercontent.com/8364014/36645218-536f7626-1a33-11e8-8677-e576580c1e37.PNG)

But when I start swiping in the container view it readjusts and looks normal:
![img_0981](https://user-images.githubusercontent.com/8364014/36645219-56c8fcc0-1a33-11e8-81c8-fbc6a4a8856d.PNG)

Really not sure what's going on here, any suggestions are very welcome!

**Comment by tooolbox** [comment]
@miranda-elliott The only thing I could think to suggest would be an extra `reload` call in there to try to jank it into place, on `viewDidLoad` or `viewWillAppear` or something, but that's obviously not solving the root of the problem.

Good luck :)

**Comment by miranda-elliott** [comment]
Ended up scouring ButtonBarPagerTabStripViewController.swift for any height values and found that it's setting the buttonBarHeight on line 116 with `let buttonBarHeight = settings.style.buttonBarHeight ?? 44 `. Set my collection view's height constraint to 44 and it's no longer throwing the error. Definitely think that spec should be specified in the readme or tutorial somewhere as the default value, but glad I solved it regardless!
